app/scripts/components/sandbox/mdxEditor/MDX.tsx

```diff
@@ -0,0 +1,264 @@
+                }}
+              >
+                reset with default content
+              </ResetButton>
+              <ResetButton
+                type='button'
+                onClick={() => {
+                  setMdxSource(DEFAULT_CONTENT);
+                }}
+              >
+                clear
+              </ResetButton>
+            </TitleBar>
+            {error && <ErrorBar>{errorHumanReadable}</ErrorBar>}
+            <EditorWrapper>
+              <CodeMirror
+                value={initialSource}
+                onUpdate={(v) => {
+                  if (v.docChanged) {
+                    let source = v.state.doc.toString();
+
+                    // This is a hack to prevent the editor from being cleared
+                    // when the user deletes all the text.
+                    // because when that happens, React throws an order of hooks error
+                    source = source ? source : DEFAULT_CONTENT;
+                    setSource(source);
+                    setMdxSource(source);
+                  }
+                }}
+                extensions={extensions}
+              />
+            </EditorWrapper>
+          </DraggableEditor>
+        </Draggable>
+        <MDXRenderer result={result} components={components} />
+      </MDXContext.Provider>
+    </ErrorBoundaryWithCRAReset>
+  );
+};
+
+class ErrorBoundaryWithCRAReset extends ErrorBoundary {
+  static getDerivedStateFromError(error: Error) {
+    (error as any).CRAOverlayIgnore = true;
+    return { didCatch: true, error };
+  }
+}
+
+interface MDXRendererProps {
+  result: MDXContent | null;
+  components: any;
+}
+
+const MDXRenderer = ({ result, components }: MDXRendererProps) => {
+  return (
+    <MDXProvider components={components}>
+      {result && result({ components })}
+    </MDXProvider>
+  );
+};
+
+const GlobalError = () => {
+  const [, setMdxSource] = useLocalStorage(
+    MDX_LOCAL_STORAGE_KEY,
+    MDX_SOURCE_DEFAULT
+  );
+  return (
+    <GlobalErrorWrapper>
+      An error occurred
+      <div>
+        <button
+          type='button'
+          onClick={() => {
+            window.location.reload();
+          }}
+        >
+          Try again with recovered content
+        </button>
+        <button
+          type='button'
+          onClick={() => {
+            setMdxSource(MDX_SOURCE_DEFAULT);
+            window.location.reload();
+          }}
+        >
+          Try again with content reset to defaults
+        </button>
+      </div>
+    </GlobalErrorWrapper>
+  );
+};
+
+const MDXBlockError = ({ error }: any) => {
+  return (
+    <HintedErrorDisplay
+      title={generalErrorMessage}
+      subtitle={docsMessage}
+      message={error.message}
+      hints={error.hints}
+    />
+  );
+};
+
+export const MDXBlockWithError = (props) => {
+  const result = useContext(MDXContext);
+  return (
+    <ErrorBoundaryWithCRAReset
+      FallbackComponent={MDXBlockError}
+      resetKeys={[result]}
```

Comment:
❗ resetKeys ❗

Reply:
Anything wrong here?

Reply:
What is this `resetKeys`? I haven't seen it being used anywhere

Reply:
@hanbyul-here @danielfdsilva, indeed it's part of `react-error-boundary`'s API. There's not much documentation, but this [blog post](https://kentcdodds.com/blog/use-react-error-boundary-to-handle-errors-in-react) explains it, kinda works like a memoize.

Reply:
ahh, Thank you! That's an interesting feature.
I had looked at the docs, but didn't find it. It's a bit buried. 🫣